connectors/src/connectors/microsoft/index.ts

```diff
@@ -363,9 +363,13 @@ export class MicrosoftConnectorManager extends BaseConnectorManager<null> {
         internalId: id,
       }));
 
-    await MicrosoftRootResource.batchMakeNew(newResourcesBlobs);
+    const addedResources =
+      await MicrosoftRootResource.batchMakeNew(newResourcesBlobs);
```

Comment:
if we just change the front end logic so here we only pass the 'changed' permissions, wouldn't that and the `activities.ts` changes below be enough?

Reply:
taking care of removed folders too ofc

Reply:
The removal is handled same way as before (deletion workflow is launched before starting new workflow/sending signal)